src/lib/swapper/swappers/ThorchainSwapper/utils/getL1quote.ts

```diff
@@ -221,14 +222,24 @@ export const getL1quote = async (
             const buyAmountBeforeFeesCryptoBaseUnit =
               getRouteBuyAmountBeforeFeesCryptoBaseUnit(quote)
 
-            const updatedMemo = addSlippageToMemo({
+            const limitWithManualSlippage = getLimitWithManualSlippage({
               expectedAmountOutThorBaseUnit,
-              quotedMemo: quote.memo,
               slippageBps,
-              chainId: sellAsset.chainId,
-              affiliateBps,
-              isStreaming,
             })
+
+            if (!quote.memo) throw new Error('no memo provided')
+
+            let updatedMemo = quote.memo
+
+            // always use TC auto stream quote (0 limit = 5bps - 50bps, sometimes up to 100bps)
+            // see: https://discord.com/channels/838986635756044328/1166265575941619742/1166500062101250100
+            if (!isStreaming) {
+              updatedMemo = addLimitToMemo({
+                memo: quote.memo,
+                limit: limitWithManualSlippage,
+              })
+            }
+
```

Comment:
stylistic (non-blocking): use an IIFE/ternary instead of let syntax and mutating e.g:

```typescript
            // always use TC auto stream quote (0 limit = 5bps - 50bps, sometimes up to 100bps)
            // see: https://discord.com/channels/838986635756044328/1166265575941619742/1166500062101250100
            const memo = isStreaming
              ? quote.memo
              : addLimitToMemo({ memo: quote.memo, limit: limitWithManualSlippage })
```

And the same for the other occurrences in l.335 and l.450

Reply:
Trust me or not but initially I did this ternary and I switched back to the mutating pattern after telling myself "gomes will told me that this ternary makes it harder to read"

Done in [ad907d8](https://github.com/shapeshift/web/pull/7138/commits/ad907d8b9e045937eefbd18632d551a76a6b7819)